server/sandbox_stop.go

```diff
@@ -51,10 +53,38 @@ func (s *Server) StopPodSandbox(ctx context.Context, req *pb.StopPodSandboxReque
 			if err := s.runtime.StopContainer(c, -1); err != nil {
 				return nil, fmt.Errorf("failed to stop container %s in pod sandbox %s: %v", c.Name(), sb.id, err)
 			}
+			if c.ID() != podInfraContainer.ID() {
+				if err := s.storageRuntimeServer.StopContainer(c.ID()); err != nil {
+					return nil, fmt.Errorf("failed to stop container %s in pod sandbox %s: %v", c.Name(), sb.id, err)
+				}
+			}
 		}
 		s.containerStateToDisk(c)
 	}
 
+	if sb.cleanupDone {
+		resp := &pb.StopPodSandboxResponse{}
+		logrus.Debugf("StopPodSandboxResponse: %+v", resp)
+		return resp, nil
+	}
+
+	if err := label.ReleaseLabel(sb.processLabel); err != nil {
+		return nil, err
+	}
+
+	// unmount the shm for the pod
+	if sb.shmPath != "/dev/shm" {
+		if err := syscall.Unmount(sb.shmPath, syscall.MNT_DETACH); err != nil {
+			return nil, err
+		}
+	}
+
+	if err := s.storageRuntimeServer.StopContainer(sb.id); err != nil {
```

Comment:
The cleanupDone state variable stinks, I'll remove it and make sure this is idempotent (and add a Stat of the shm mount before trying to umounting it. Hold on the review.